Rename "vertex" to "control point" across the class reference

Shaderd00d opened the pull request:
Renamed 'vertex' to 'curve point' in the Curve2D and Curve3D class documentation because curves don't have vertices.

Shaderd00d committed:
Renamed vertex in Curve2D and Curve3D class docs

Renamed 'vertex' to 'curve point' in the Curve2D and Curve3D class documentation because curves don't have vertices

fire commented:
I've heard of it called control points, and it's called knots or edit points if the control point is on the surface of the curve.

Shaderd00d committed:
Using the term control point in the Curve2D class docs

Shaderd00d committed:
Using the term control point in the Curve3D class docs

Shaderd00d committed:
Fixed typo in Curve2D class docs

Shaderd00d commented:
Thanks, I renamed it.

Shaderd00d committed:
Renamed vertex in PathFollow2D/3D class docs

Renamed vertex to point in the PathFollow2D and PathFollow3D class documentation because paths don't have vertices.

Mickeon commented on doc/classes/Curve2D.xml:
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8" ?>
+﻿<?xml version="1.0" encoding="UTF-8" ?>
```
Whatever program you used did some sneaky reformatting in this line for both Curve2D and Curve3D which causes the checks to fail. They need to be reverted.

Shaderd00d committed:
Reverted XML header

Shaderd00d commented on doc/classes/Curve2D.xml:
```diff
@@ -59,21 +59,21 @@
 			<param index="0" name="idx" type="int" />
 			<description>
-				Returns the position of the control point leading out of the vertex [param idx]. The returned position is relative to the vertex [param idx]. If the index is out of bounds, the function sends an error to the console, and returns [code](0, 0)[/code].
+				Returns the position of the control point leading out of the control point [param idx]. The returned position is relative to the control point [param idx]. If the index is out of bounds, the function sends an error to the console, and returns [code](0, 0)[/code].
```
Curve2D and Curve3D use the [bezier_interpolate()](https://docs.godotengine.org/en/stable/classes/class_vector2.html#class-vector2-method-bezier-interpolate) function of Vector2 and Vector3.
The description of the function also uses the term "control points" and links to a [Bézier curve article on Wikipedia](https://en.wikipedia.org/wiki/B%C3%A9zier_curve) where it's used, too. I couldn't find the term "vertex" in this context and think "control point" is the correct replacement.

tetrapod00 replied:
Right, "control point" might be a fine substitute for some of these usages. But look at the existing description in this case:
> Returns the position of the control point leading out of the vertex [param idx]...

And compare it to your changed version:
> Returns the position of the control point leading out of the control point [param idx]...

The term "control point" is already used. But it's not used for every kind of point, because the existing description draws a distinction between "control points" and "vertices". So at the very least in this case, I'm not sure if substituting "control point" for "vertex" is correct. And in all the other changed cases, you should be careful you're not introducing a subtle error.

Mickeon replied:
Maybe I'm repeating the exact same sentiment as @tetrapod00, but [all points of a curve are referred to as "control points"](https://en.wikipedia.org/wiki/B%C3%A9zier_curve#:~:text=The%20points%20Pi%20are,polygon%20contains%20the%20B%C3%A9zier%20curve.). However, it seems like the existing descriptions hint that only `in` and `out` should be named that way, which is not accurate as the `position` is a control point, too.

I think repeating the same word twice is confusing, but I have little suggestions. Perhaps `set_point_in` and similar should not repeat "control" twice.

tetrapod00 replied:
The problem might run deeper, actually. The [class description](https://docs.godotengine.org/en/latest/classes/class_curve2d.html#description) for Curve2D describes it as "a Bézier curve in 2D space". But is that accurate? You can (manually) change the control points coming out of each "Point"/"vertex" so that some "vertices" are sharp:

![Godot_v4 3-stable_mono_win64_83PMOSIntD](https://github.com/user-attachments/assets/7c833321-cce7-467c-9a68-dd75eb39abc0)

To me, this looks more like a [Composite Bezier Curve](https://en.wikipedia.org/wiki/Composite_B%C3%A9zier_curve) as defined by Wikipedia, which is a *series* of individual Bezier curves. In that case, "vertex" is not a totally unused term, going by Wikipedia:
> A béziergon (also called bézigon) is a closed path composed of [Bézier curves](https://en.wikipedia.org/wiki/B%C3%A9zier_curve). It is similar to a [polygon](https://en.wikipedia.org/wiki/Polygon) in that it connects a set of [vertices](https://en.wikipedia.org/wiki/Vertex_(geometry)) by lines, but whereas in polygons the vertices are connected by straight lines, in a béziergon the vertices are connected by Bézier curves.

The method description for [add_point()](https://docs.godotengine.org/en/latest/classes/class_curve2d.html#class-curve2d-method-add-point) calls the "vertices" simply "points", which have a "position" and "control points":
> Adds a point with the specified `position` relative to the curve's own position, with control points `in` and `out`. Appends the new point at the end of the point list.

That `position` *will* be used as a control point for the individual Bezier curves which connect the "points" or "vertices". But currently there is a distinction drawn between that control point, and the `in` and `out` control points, because they are edited in different ways. The Curve2D class is *not* simply a list of control points which make a single Bezier curve. I don't think it makes sense to call everything here a control point because you lose some clarity.

But, I really don't have a lot of experience with curves and the precise terminology, though.

Mickeon replied:
No, these are likely incorrect if only for the fact that they were originally written in 8abdb43aabd4f77bd33f412a9b13eec4e02d67e0 back when very little reviewing was done, if I may be so frank.

Shaderd00d replied:
A Composite Bezier Curve is continuous according to the Wikipedia site, but it appears to be close to what we have.

Looking at the [code](https://github.com/godotengine/godot/blob/2582793d408ade0b6ed42f913ae33e7da5fb9184/scene/resources/curve.cpp#L822), a Curve2D is sampled with 4 points like a cubic Bezier curve. While following this hint, I found this information about [specific cases](https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Specific_cases):

> A Bézier curve is defined by a [set](https://en.wikipedia.org/wiki/Set_(mathematics)) of [control points](https://en.wikipedia.org/wiki/Control_point_(mathematics)) P0 through Pn, where n is called the order of the curve (n = 1 for linear, 2 for quadratic, 3 for cubic, etc.). The first and last control points are always the endpoints of the curve; however, the intermediate control points generally do not lie on the curve. The sums in the following sections are to be understood as [affine combinations](https://en.wikipedia.org/wiki/Affine_space#Affine_combinations_and_barycenter) – that is, the coefficients sum to 1.

Is this what we have in Godot, using "control points"?

I'm at a loss to be honest...

fire replied:
Is the conclusion to change or keep the existing terminology?

Shaderd00d replied:
I'll close this PR as it probably makes sense to leave it as it is. Thanks for all the comments and efforts,
